### PR TITLE
Simplify name of generated exception mappers

### LINKED ISSUE #43786: Rest: shorten the names of the generated classes
### Description

There seems to be no reason for the names to be so looong.

For example `io.quarkus.resteasy.reactive.server.runtime.exceptionmappers.AuthenticationFailedExceptionMapper$GeneratedExceptionHandlerFor$AuthenticationFailedException$OfMethod$handle`. The simple name has 111 chars in this particular case.

The  `ArC > Beans` view in the Dev UI does not look good. Also every call site of this class must store this name in the constant pool ;-).

Zulip discussion: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Loooong.20names.20of.20generated.20classes.20in.20resteasy-reactive

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
- Closes: #43786

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java [hunk 1]
```diff
@@ -645,9 +646,8 @@ private static void checkModifiers(MethodInfo info) {
     }
 
     private static String getGeneratedClassName(MethodInfo targetMethod, Type handledExceptionType) {
-        return targetMethod.declaringClass().name() + "$GeneratedExceptionHandlerFor$"
-                + handledExceptionType.name().withoutPackagePrefix()
-                + "$OfMethod$" + targetMethod.name();
+        return targetMethod.declaringClass().name() + "$ExceptionMapper$"
```
Comment (FroMage) [inline independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java:649]: Why not replace `$ExceptionMapper$` with the method name?

**Review by FroMage — APPROVED** [review]

File: independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java (hunk 1 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java:649]: I find that looking at this name I know for sure that is a generated exception mapper without having to look into the code itself

Comment (FroMage) [reply independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java:649]: Indeed you know what it is, but you don't know where it's defined, so you can't find it back easily.

Comment (FroMage) [reply independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java:649]: Whereas if you can find it back, then it's pretty obvious what it is once you look at it.

**Pull request merged**